src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -448,7 +453,7 @@ private void EliminateEndingBacktracking()
                     // an Atomic one if its grandparent is already Atomic.
                     // e.g. [xyz](?:abc|def) => [xyz](?>abc|def)
                     case RegexNodeKind.Capture:
-                    case RegexNodeKind.Concatenate:
+                    case RegexNodeKind.Concatenate when !rtl:
```

Comment:
can both your two rtl tests in this method be hit by tests?

Reply:
Yup